addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -219,6 +228,80 @@ odoo.define('website_slides.quiz', function (require) {
             });
         },
 
+        /**
+         * @private
+         */
+        _onClicksigninJoin: function (event) {
+            event.preventDefault();
+            var self = this;
+            var values = self._getAnswers()
+            if (values.length === self.quiz.questions.length){
+                self._alertHide();
+                if (this.publicUser){
+                    values = {'slide_id': this.slide.id, 'slide_answers':values}
+                    return this._rpc({
+                        route:'/slides/slide/quiz/save_slide_answsers',
+                        params: {
+                            'slide_values': values,
+                        }
+                    }).then(function (){
+                        var url = self._createLoginRedirectUrl()
+                        window.location.href = url;
+                    });
+                }
+            } else {
+                self._alertShow();
+            }
+        },
+        /**
+         * @private
+         * @param
+         */
+        _createLoginRedirectUrl: function(){
+            var urlParts = window.location.pathname.split('/');
+            var url =  urlParts.join('/');
+            var params = {};
+            if (window.location.href.indexOf("fullscreen") > -1) {
+                params.fullscreen=1;
+            }
+            this.redirectURL= _.str.sprintf('%s?%s', url, $.param(params));
+            return _.str.sprintf('/web/login?redirect=%s', encodeURIComponent(this.redirectURL));
+        },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+        _alertHide: function () {
+            this.$('.o_wslides_js_lesson_quiz_error').addClass('d-none');
+        },
+
+        _alertShow: function (alert_code) {
+            var message = _t('There was an error validating this quiz.');
+            if (! alert_code || alert_code === 'slide_quiz_incomplete') {
+                message = _t('All questions must be answered !');
+            }
+            else if (alert_code === 'slide_quiz_done') {
+                message = _t('This quiz is already done. Retaking it is not possible.');
+            }
+            else if (alert_code === 'public_user') {
+                message = _t('You must be logged to submit the quiz.');
+            }
+            this.$('.o_wslides_js_lesson_quiz_error span:first').html(message);
+            this.$('.o_wslides_js_lesson_quiz_error').removeClass('d-none');
+        },
+        /**
+         * Get the answers filled in by the User
+         *
+         * @private
+         * @param 
+         */
+        _getAnswers: function () {
+            var inputs = this.$('input[type=radio]:checked');
```

Comment:
You should be able to use `.map()` to reduce code here.
Something like:

`return this.$('input[type=radio]:checked').map(function (element) {return $(element).val();});`

To check/test

Reply:
```

return jQuery.makeArray(this.$('input[type=radio]:checked').map(function (index, element) {
                return [$(element).val()];
            }));
```
It is working if I do this.

Reply:
Can you try `return this.$('input[type=radio]:checked').map(function (element) {return $(element).val();}).get();` ?

The important difference being the `get`, sorry I forgot to put it in my first comment.

Reply:
this works:
```
return this.$('input[type=radio]:checked').map(function (index, element) {
    return parseInt($(element).val());
 }).get();
```

I need the parseint because otherwise it returns an array of strings and I work with ints later on and removing the 'index' argument of the function in the map causes an error.

Reply:
Already a lot better, thanks for experimenting with it!